lib/rdoc/markup/pre_process.rb

```diff
@@ -220,11 +223,11 @@ def handle_directive prefix, directive, param, code_object = nil,
       # remove parameter &block
       code_object.params = code_object.params.sub(/,?\s*&\w+/, '') if code_object.params
 
-      code_object.block_params = param
+      code_object.block_params = param || ''
 
       blankline
     else
-      result = yield directive, param if block_given?
+      result = yield directive, param, line if block_given?
```

Comment:
What are these changes for? Looks like they're fixing issues that'd also affect the default Ruby parser?

Reply:
In the default ruby parser, the third parameter `line` is just ignored.

The line where `:method:` is written is the source line number of ghost method.
```ruby
1| class A
2|   ##
3|   # this is a ghost method `foo(x, y). Source location is line 4
4|   # :method: foo
5|   # :args: x, y
6| end
```

The original code is calculating line number by this code.
https://github.com/ruby/rdoc/blob/1bb3becd00704de879343eec31d202cf5e665862/lib/rdoc/parser/ruby.rb#L1094-L1095

`pre_process.rb` parses comment directives, `parse_comment` also parses some comment directives.
To avoid re-implementing these kind of thing, I changed `PreProcess#handle` to also provide line_number for each directive.

This is not the best way. pre_process has problem and I think it should be fixed in the future. pre_process needs code_object already created, but we need to parse the comment first to identify code_object type which can be either attribute or method.